### PR TITLE
Retreives Config from a local image using the docker command

### PR DESCRIPTION
**Pull request by patrick-premont**
Created CmdConfig because Config didn't parse exactly right: the config field had to start with an uppercase.

Refactored the invocation of getDockerHostConfig and getDockerRegistryConfig to be more concise, so that adding a third method produced a more readable sequence.

### CODE REVIEW

**Comment by patrick-premont** [comment]
This helps with builds where no remote docker registry is available, or where authentication is challenging. It also is much faster (less than 0.5s). For me a run with docker-credentials-gcloud) took at least 13 seconds.

File: cli/shared/src/main/scala/com/lightbend/rp/reactivecli/Main.scala [hunk 1]
```diff
@@ -191,6 +191,13 @@ object Main extends LazyLogging {
                   }
                 }
 
+                def getDockerCmdConfig(imageName: String): Future[Option[Config]] =
+                  Platform.processExec(Seq("docker", "inspect", imageName)).map {
```
Comment (mitkus) [inline cli/shared/src/main/scala/com/lightbend/rp/reactivecli/Main.scala]: Executing docker command should be wrapped into a separate object with debug logging, like other native tools that cli uses - jq, kubectl, docker-credentials.

File: cli/shared/src/main/scala/com/lightbend/rp/reactivecli/docker/CmdConfig.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+package com.lightbend.rp.reactivecli.docker
+
+import argonaut.CodecJson
+import com.lightbend.rp.reactivecli.docker.{ Config => RegistryConfig }
+
+case class CmdConfig(Config: RegistryConfig.Cfg) {
+  def registryConfig: RegistryConfig = RegistryConfig(Config)
+}
+
+object CmdConfig {
```
Comment (mitkus) [inline cli/shared/src/main/scala/com/lightbend/rp/reactivecli/docker/CmdConfig.scala]: Probably better to limit scope of this inside a helper object for invoking docker. Took me a minute to figure out what's actually happening here as the name doesn't tell much!

**Comment by patrick-premont** [comment]
Kubernetes is fairly explicit when it fails to retrieve an image, so I'm not sure much confusion is possible. Remote registry authentication issues are much more complex.

Even if rp can access the image, from a developer machine or CI agent, it doesn't guarantee that the image will be accessible from the Kubernetes cluster (possibly a different network, auth).

Also, the performance benefits favor making local the default.

File: cli/shared/src/main/scala/com/lightbend/rp/reactivecli/docker/CmdConfig.scala (hunk 2 above)
Comment (patrick-premont) [reply cli/shared/src/main/scala/com/lightbend/rp/reactivecli/docker/CmdConfig.scala]: I followed the pattern set by `SocketConfig`, which might have the same issue. Agreed I'll restrict the scope.

File: cli/shared/src/main/scala/com/lightbend/rp/reactivecli/Main.scala (hunk 1 above)
Comment (patrick-premont) [reply cli/shared/src/main/scala/com/lightbend/rp/reactivecli/Main.scala]: Will do.

**Comment by dbrinegar** [comment]
Totally get the need to free up workflow.  I think the idea under this aspect of the rp tool is that it connects the dots correctly, not so much to prevent obvious errors like "container not found" but non-obvious errors like the container having the same name but wrong inner bits needed, where it would seem to start but isn't working as hoped.

I like the idea of putting this "use local, I'll take care of it" workflow behind a flag because there's more burden on the developer to make sure remote is correct.  The flag is a nice way to document that gap.

But, at the same time, you know, there are moving parts here which perhaps don't need to be so bolted onto each other like this.  And perhaps we can refactor things to make this kind of end to end validation more on the start-app side, so the build-app side doesn't have to go look around at all.

**Comment by patrick-premont** [comment]
I see. Concretely I guess the risk is only that you `docker:publishLocal`, run `rp`, and forget to `docker:publish` (push), when you had done a previous push for the same version number. This is not really a risk for CI, as distinct version numbers are used. This might occur for developers, but if a user does that and we lookup remote registries in rp, they will still end up redeploying their previous yaml. Admittedly they might notice more easily. I'm not convinced this small risk is worth the hassle of auth and the lower performance.

**Comment by edwardcallahan** [comment]
Patrick's comment speaks to the difference in `publish` vs `publishLocal` workflows. The continuation of the workflow choice is the docker image string provided to `rp`. 

Should local registry generation generate a warning? If the hostname is local/none warn that the config won't work elsewhere.

**Comment by patrick-premont** [comment]
I've moved the feature behind a flag. That's good enough for me. I guess we could deprecate it if we were to eventually agree on making it the default behavior.

**Pull request merged**